magellan-library/src/main/java/com/wealthfront/magellan/navigation/LazySetNavigator.kt

```diff
@@ -36,9 +39,39 @@ public open class LazySetNavigator(
   }
 
   public fun addNavigable(navigable: NavigableCompat) {
+    existingNavigables.add(navigable)
     lifecycleRegistry.attachToLifecycleWithMaxState(navigable, LifecycleLimit.CREATED)
   }
 
+  public fun removeNavigables(navigables: Set<NavigableCompat>) {
+    for (navigable in navigables) {
```

Comment:
Is there a risk of `ConcurrentModificationException` if one of the navigables tries to add / remove a navigable in its `onDestroy` ? I don't know enough about `mutableSetOf` to know if it allows concurrent modification